### ISSUE TITLE
L1T TSG Integration Branch based on CMSSW_8_0_2

### ISSUE THREAD

**Issue by mulhearn** [cms-l1t-offline/cmssw#191]
The latest L1T code has been ported successfully to CMSSW 802.
https://github.com/cms-l1t-offline/cmssw/tree/l1t-tsg-CMSSW_8_0_2
(This replaces the old "dev-recipes" branch at issue #147)
For stability, this branch is merely a delayed clone of the l1t private integration branch (#189) but only contains updates after they have been checked by L1T DPG.

The last tag is l1t-tsg-v5-cand.  The l1t rates plots are still pending, and
are expected not to show issues. 

For various work-flows related using the latest TSG tag, see the Twiki here:
  https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideL1TStage2Instructions

The recipe for compiling the latest stable tag is:

```
cmsrel CMSSW_8_0_2
cd CMSSW_8_0_2/src
cmsenv
git cms-init
git remote add cms-l1t-offline git@github.com:cms-l1t-offline/cmssw.git
git fetch cms-l1t-offline
git cms-merge-topic cms-l1t-offline:l1t-tsg-v5-cand
git cms-addpkg L1Trigger/L1TCommon
scram b -j 8
```

The latest tags are:
- l1t-tsg-v5-cand: equivalent to l1t-integration-v37.2
- l1t-tsg-v4:  equivalent to l1t-integration-v13.1
- l1t-tsg-v3-cmssw802: equivalent to l1t-integration-v7.0-cmssw802, which is nearly bitwise equivalent to l1t-tsg-v3 but running under CMSSW_8_0_2.  The only differences have been shown to be due to updates of HCAL TPs in 802 which has a minor effect (likely to be unnoticed unless you do bitwise comparisons)

The latest control plots from a short run over 10k zero-bias events for l1t-tsg-v4 tag are here:
http://mulhearn.web.cern.ch/mulhearn/plots/l1t-integration-v13.1/

**Comment by mulhearn** [comment]
Commmand sequence tested, disclaimer removed.  @apana is updating the Twiki, and will post a note here when that disclaimer is no longer needed.

**Comment by apana** [comment]
The SWGuideL1TStage2Instructions twiki has been updated.

-Len

**Comment by blwiner** [comment]
@Mulhearn   In this branch the unpacking of the uGT is commented out.  Is
that intentional?  If so, why?

 -Brian

On Mon, Mar 14, 2016 at 7:27 AM, Leonard Apanasvich <
notifications@github.com> wrote:

> The SWGuideL1TStage2Instructions twiki has been updated.
> 
> -Len
> 
> On Mon, Mar 14, 2016 at 11:58 AM, mulhearn notifications@github.com
> wrote:
> 
> > Commmand sequence tested, disclaimer removed. @apana
> > https://github.com/apana is updating the Twiki, and will post a note
> > here when that disclaimer is no longer needed.
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > <
> > https://github.com/cms-l1t-offline/cmssw/issues/191#issuecomment-196258773
> > 
> > .
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/cms-l1t-offline/cmssw/issues/191#issuecomment-196267965
> .

## 

# 

| Prof. Brian L. Winer |
| --- |

```
              Department of Physics
           The Ohio State University
             191 W. Woodruff Ave
              Columbus, OH 43210

               (614) 292-8996 (Office)
               (614) 292-7370 (Lab)
```

#

**Comment by blwiner** [comment]
...to clarify a little more.  Unpacking of the _inputs_ are commented out.

https://github.com/cms-l1t-offline/cmssw/blob/l1t-tsg-CMSSW_8_0_2/EventFilter/L1TRawToDigi/src/implementations_stage2/GTSetup.cc#L72

On Mon, Mar 14, 2016 at 8:31 AM, Brian Winer winer@physics.osu.edu wrote:

> @Mulhearn   In this branch the unpacking of the uGT is commented out.  Is
> that intentional?  If so, why?
> 
>  -Brian
> 
> On Mon, Mar 14, 2016 at 7:27 AM, Leonard Apanasvich <
> notifications@github.com> wrote:
> 
> > The SWGuideL1TStage2Instructions twiki has been updated.
> > 
> > -Len
> > 
> > On Mon, Mar 14, 2016 at 11:58 AM, mulhearn notifications@github.com
> > wrote:
> > 
> > > Commmand sequence tested, disclaimer removed. @apana
> > > https://github.com/apana is updating the Twiki, and will post a note
> > > here when that disclaimer is no longer needed.
> > > 
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <
> > > https://github.com/cms-l1t-offline/cmssw/issues/191#issuecomment-196258773
> > > 
> > > .
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/cms-l1t-offline/cmssw/issues/191#issuecomment-196267965
> > .
> 
> ## 
> 
> # 
> 
> | Prof. Brian L. Winer |
> | --- |
> 
> ```
>               Department of Physics
>            The Ohio State University
>              191 W. Woodruff Ave
>               Columbus, OH 43210
> 
>                (614) 292-8996 (Office)
>                (614) 292-7370 (Lab)
> ```
> 
> # 

## 

# 

| Prof. Brian L. Winer |
| --- |

```
              Department of Physics
           The Ohio State University
             191 W. Woodruff Ave
              Columbus, OH 43210

               (614) 292-8996 (Office)
               (614) 292-7370 (Lab)
```

#

**Comment by fwyzard** [comment]
@mulhearn , @rekovic , can you make a PR for CMSSW 8.0.x and 8.1.x ?

**Comment by mulhearn** [comment]
Yes, will do, just need to clean up the XXX/data/...

**Comment by mulhearn** [comment]
With apologies to @apana , for expecting him to read my mind, the recipe for HF 1x1 TPs has changed in 802:

A) For MC in 802, you should use GT 80X_mcRun2_asymptotic_v6 and drop
the HF 1x1 TP event setup customization
(i.e. omit "--customise=L1Trigger/Configuration/customiseReEmul.L1TEventSetupForHF1x1TPs")

B) For data in 802, no change from recipe is needed, in particular,
you should continue using
"--customise=L1Trigger/Configuration/customiseReEmul.L1TEventSetupForHF1x1TPs"
which has been updated to include conditions needed for 802.  I should
note that the HCAL folks recommended data tag "80X_dataRun2_HLT_v6"
but as we are re-emulating over 2015D, the conditions override is
anyway still needed.  So we are just leaving the data recipe intact,
included using "auto:run2_data" conditions.

**Comment by mulhearn** [comment]
@apana:  I just pushed tag l1t-integration-v14.1 which adds a customization to run a bitwise Layer1 instance in parallel, saving results to ntuple, as in this demo:
`cmsDriver.py L1TEST --python_filename=l1t_test.py -s RAW2DIGI --era=Run2_2016 --customise=L1Trigger/Configuration/customiseReEmul.L1TReEmulFromRAW --customise=L1Trigger/L1TNtuples/customiseL1Ntuple.L1NtupleEMU --customise=L1Trigger/Configuration/customiseUtils.L1TTurnOffUnpackStage2GtGmtAndCalo --conditions=auto:run2_data -n 100 --data --no_exec --no_output --filein=/store/data/Run2015D/MuonEG/RAW/v1/000/256/677/00000/4A874FB5-585D-E511-A3D8-02163E0143B5.root --geometry=Extended2016,Extended2016Reco --customise=L1Trigger/Configuration/customiseReEmul.L1TEventSetupForHF1x1TPs --customise=L1Trigger/Configuration/customiseReEmul.L1TAddBitwiseLayer1`
